src/lib.rs

```diff
@@ -1032,79 +1032,156 @@ pub struct Keys<'a, K: 'a, V: 'a> {
 impl<'a, K, V> Iterator for Keys<'a, K, V> {
     type Item = &'a K;
 
+    #[inline]
```

Comment:
I would prefer to not specify this, it's the compiler's job. The item is already inlinable, so we don't need to do more. Same for all the other methods in the PR